include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,988 @@
+     *value found until that generation, Kernel is the kernel size, Oracle is the oracle parameter value, dx is the
+     *flatness in the individuals, dp is the flatness in the penalty function values.
+     *
+     * @param level verbosity level
+     */
+    void set_verbosity(unsigned level)
+    {
+        m_verbosity = level;
+    }
+    /// Gets the verbosity level
+    /**
+     * @return the verbosity level
+     */
+    unsigned get_verbosity() const
+    {
+        return m_verbosity;
+    }
+    /// Gets the generations
+    /**
+     * @return the number of generations to evolve for
+     */
+    unsigned get_gen() const
+    {
+        return m_gen;
+    }
+    /// Algorithm name
+    /**
+     * Returns the name of the algorithm.
+     *
+     * @return <tt> std::string </tt> containing the algorithm name
+     */
+    std::string get_name() const
+    {
+        return "GACO: Ant Colony Optimization";
+    }
+    /// Extra informations
+    /**
+     * Returns extra information on the algorithm.
+     *
+     * @return an <tt> std::string </tt> containing extra informations on the algorithm
+     */
+    std::string get_extra_info() const
+    {
+        std::ostringstream ss;
+        stream(ss, "\tGenerations: ", m_gen);
+        stream(ss, "\n\tAccuracy parameter: ", m_acc);
+        stream(ss, "\n\tImprovement stopping criterion: ", m_impstop);
+        stream(ss, "\n\tEvaluation stopping criterion: ", m_evalstop);
+        stream(ss, "\n\tFocus parameter: ", m_focus);
+        stream(ss, "\n\tKernel: ", m_ker);
+        stream(ss, "\n\tOracle parameter: ", m_oracle);
+        stream(ss, "\n\tMax number of non-dominated solutions: ", m_paretomax);
+        stream(ss, "\n\tPareto precision: ", m_epsilon);
+        stream(ss, "\n\tPseudo-random number generator (Marsenne Twister 19937): ", m_e);
+        stream(ss, "\n\tSeed: ", m_seed);
+        stream(ss, "\n\tVerbosity: ", m_verbosity);
+
+        return ss.str();
+    }
+    /// Get log
+    /**
+     * A log containing relevant quantities monitoring the last call to evolve. Each element of the returned
+     * <tt>std::vector</tt> is a gaco::log_line_type containing: gen, m_fevals, best_fit, m_ker,
+     * m_oracle, dx, dp
+     * as described in gaco::set_verbosity
+     * @return an <tt>std::vector</tt> of gaco::log_line_type containing the logged values gen, m_fevals,
+     * best_fit, m_ker, m_oracle, dx, dp
+     */
+    const log_type &get_log() const
+    {
+        return m_log;
+    }
+    /// Object serialization
+    /**
+     * This method will save/load \p this into the archive \p ar.
+     *
+     * @param ar target archive.
+     *
+     * @throws unspecified any exception thrown by the serialization of the UDP and of primitive types.
+     */
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(m_gen, m_acc, m_impstop, m_evalstop, m_focus, m_ker, m_oracle, m_paretomax, m_epsilon, m_e, m_seed,
```

Comment:
I count 24 data members in this class, but it seems like only 23 of them are serialized. Did you forget some members in this call?

Reply:
@darioizzo 
Indeed there was a missing data member (`m_sol_archive`) and I added it  in the commit.